x/swaprouter/router.go

```diff
@@ -15,15 +18,131 @@ func (k Keeper) RouteExactAmountIn(
 	routes []types.SwapAmountInRoute,
 	tokenIn sdk.Coin,
 	tokenOutMinAmount sdk.Int) (tokenOutAmount sdk.Int, err error) {
-	panic("not implemented")
+	var (
```

Comment:
Note: confirmed that this is copied with minimal changes.

The only additional change being choosing the module to route to